File: Lib/Telerik/Examples/QuickStart/Forms/AboutBox/RadAboutBox1.vb
Imports System
Imports System.Drawing
Imports System.IO
Imports System.Reflection
Imports Telerik.QuickStart.WinControls
Imports Telerik.WinControls

Namespace Telerik.Examples.WinControls.Forms.AboutBox
    Partial Class RadAboutBox1
        Inherits ExamplesRadForm

        Private executingAssembly_field As Assembly

        Public Sub New()
            InitializeComponent()
            Me.Text = String.Format("About {0}", Me.AssemblyTitle)
            Me.radLabelProductName.Text = Me.AssemblyProduct
            Me.radLabelVersion.Text = String.Format("Version {0}", Me.AssemblyVersion)
            Dim assemblyCopyright As String = Me.AssemblyCopyright
            Me.radLabelCopyright.Text = assemblyCopyright.Substring(0, assemblyCopyright.IndexOf("."))
            Me.radLabelCompanyName.Text = Me.AssemblyCompany
            Me.radTextBoxDescription.Text = VersionNumber.ProductName & " includes over 140 UI controls that you can " & "use to easily build unique and visually stunning Line of Business applications. " & "Boasting a well-designed architecture and CAB support, RadControls are perfectly " & "suited for your Enterprise desktop development."
            Me.logoPictureBox.Image = Me.GetAboutBoxImage()
        End Sub

        Public ReadOnly Property ExecutingAssembly As Assembly
            Get
                If Me.executingAssembly_field Is Nothing Then
                    Me.executingAssembly_field = Assembly.GetExecutingAssembly()
                End If

                Return Me.executingAssembly_field
            End Get
        End Property

        Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            Dim themeName As String = Me.ThemeName
            MyBase.OnLoad(e)
            ThemeResolutionService.ApplyThemeToControlTree(Me, themeName)
        End Sub

        Private Function GetAboutBoxImage() As Image
            Dim img As Image
            Dim imageStream As Stream
            Dim imageFileName As String = "AboutBox.png"
            imageStream = Me.ExecutingAssembly.GetManifestResourceStream(imageFileName)
            img = Bitmap.FromStream(imageStream)

            Return img
        End Function

        Private Function GetAssemblyAttribute(Of T)() As T
            Dim attributes As Object() = Me.ExecutingAssembly.GetCustomAttributes(GetType(T), False)

            If attributes.Length = 0 Then
                Return Nothing
            End If

            Return CType(attributes(0), T)
        End Function

        Public ReadOnly Property AssemblyTitle As String
            Get
                Dim titleAttribute As AssemblyTitleAttribute = Me.GetAssemblyAttribute(Of AssemblyTitleAttribute)()

                If titleAttribute IsNot Nothing AndAlso titleAttribute.Title <> String.Empty Then
                    Return titleAttribute.Title
                End If

                Return Path.GetFileNameWithoutExtension(Me.ExecutingAssembly.CodeBase)
            End Get
        End Property

        Public ReadOnly Property AssemblyVersion As String
            Get
                Return Me.ExecutingAssembly.GetName().Version.ToString()
            End Get
        End Property

        Public ReadOnly Property AssemblyDescription As String
            Get
                Dim attr As AssemblyDescriptionAttribute = Me.GetAssemblyAttribute(Of AssemblyDescriptionAttribute)()

                If attr Is Nothing Then
                    Return String.Empty
                End If

                Return attr.Description
            End Get
        End Property

        Public ReadOnly Property AssemblyProduct As String
            Get
                Dim attr As AssemblyProductAttribute = Me.GetAssemblyAttribute(Of AssemblyProductAttribute)()

                If attr Is Nothing Then
                    Return String.Empty
                End If

                Return attr.Product
            End Get
        End Property

        Public ReadOnly Property AssemblyCopyright As String
            Get
                Dim attr As AssemblyCopyrightAttribute = Me.GetAssemblyAttribute(Of AssemblyCopyrightAttribute)()

                If attr Is Nothing Then
                    Return String.Empty
                End If

                Return attr.Copyright
            End Get
        End Property

        Public ReadOnly Property AssemblyCompany As String
            Get
                Dim attr As AssemblyCompanyAttribute = Me.GetAssemblyAttribute(Of AssemblyCompanyAttribute)()

                If attr Is Nothing Then
                    Return String.Empty
                End If

                Return attr.Company
            End Get
        End Property
    End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/Forms/AboutBox/RadAboutBox1.cs
using System;
using System.Drawing;
using System.IO;
using System.Reflection;
using Telerik.QuickStart.WinControls;
using Telerik.WinControls;

namespace Telerik.Examples.WinControls.Forms.AboutBox
{
    partial class RadAboutBox1 : ExamplesRadForm
    {
        private Assembly executingAssembly;

        public RadAboutBox1()
        {
            InitializeComponent();

            // Initialize the AboutBox to display the product information from the assembly information.
            // Change assembly information settings for your application through either:
            // - Project -> Properties -> Application -> Assembly Information
            // - AssemblyInfo.cs
            this.Text = string.Format("About {0}", this.AssemblyTitle);
            this.radLabelProductName.Text = this.AssemblyProduct;
            this.radLabelVersion.Text = string.Format("Version {0}", this.AssemblyVersion);
            string assemblyCopyright = this.AssemblyCopyright;
            this.radLabelCopyright.Text = assemblyCopyright.Substring(0, assemblyCopyright.IndexOf("."));
            this.radLabelCompanyName.Text = this.AssemblyCompany;
            this.radTextBoxDescription.Text = 
                VersionNumber.ProductName + " includes over 140 UI controls that you can " +
                "use to easily build unique and visually stunning Line of Business applications. " +
                "Boasting a well-designed architecture and CAB support, RadControls are perfectly " +
                "suited for your Enterprise desktop development.";
            this.logoPictureBox.Image = this.GetAboutBoxImage();
        }

        public Assembly ExecutingAssembly
        {
            get
            {
                if (this.executingAssembly == null)
                {
                    this.executingAssembly = Assembly.GetExecutingAssembly();
                }

                return this.executingAssembly;
            }
        }

        protected override void OnLoad(EventArgs e)
        {
            string themeName = this.ThemeName;

            base.OnLoad(e);

            ThemeResolutionService.ApplyThemeToControlTree(this, themeName);
        }

        private Image GetAboutBoxImage()
        {
            Image img;
            Stream imageStream;
            string imageFileName = "AboutBox.png";

            imageStream = this.ExecutingAssembly.GetManifestResourceStream(
                string.Format("Telerik.Examples.WinControls.Forms.AboutBox.{0}", imageFileName));
            img = Bitmap.FromStream(imageStream);

            if (img == null)
            {
                imageStream = this.ExecutingAssembly.GetManifestResourceStream(string.Format(imageFileName));
                img = Bitmap.FromStream(imageStream);
            }

            return img;
        }

        #region Assembly Attribute Accessors

        private T GetAssemblyAttribute<T>()
        {
            object[] attributes = this.ExecutingAssembly.GetCustomAttributes(typeof(T), false);
            if (attributes.Length == 0)
            {
                return default(T);
            }

            return (T)attributes[0];
        }

        public string AssemblyTitle
        {
            get
            {                
                AssemblyTitleAttribute titleAttribute = this.GetAssemblyAttribute<AssemblyTitleAttribute>();
                if (titleAttribute != null && titleAttribute.Title != string.Empty)
                {
                    return titleAttribute.Title;
                }

                // If there was no Title attribute, or if the Title attribute was the empty string, return the .exe name
                return Path.GetFileNameWithoutExtension(this.ExecutingAssembly.CodeBase);
            }
        }

        public string AssemblyVersion
        {
            get
            {
                return this.ExecutingAssembly.GetName().Version.ToString();
            }
        }

        public string AssemblyDescription
        {
            get
            {
                AssemblyDescriptionAttribute attr = this.GetAssemblyAttribute<AssemblyDescriptionAttribute>();
                if (attr == null)
                {
                    return string.Empty;
                }

                // If there is a Description attribute, return its value
                return attr.Description;
            }
        }

        public string AssemblyProduct
        {
            get
            {
                AssemblyProductAttribute attr = this.GetAssemblyAttribute<AssemblyProductAttribute>();
                if (attr == null)
                {
                    return string.Empty;
                }

                // If there is a Product attribute, return its value
                return attr.Product;
            }
        }

        public string AssemblyCopyright
        {
            get
            {
                AssemblyCopyrightAttribute attr = this.GetAssemblyAttribute<AssemblyCopyrightAttribute>();
                if (attr == null)
                {
                    return string.Empty;
                }

                // If there is a Copyright attribute, return its value
                return attr.Copyright;
            }
        }

        public string AssemblyCompany
        {
            get
            {
                AssemblyCompanyAttribute attr = this.GetAssemblyAttribute<AssemblyCompanyAttribute>();
                if (attr == null)
                {
                    return string.Empty;
                }

                // If there is a Company attribute, return its value
                return attr.Company;
            }
        }

        #endregion
    }
}
